Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp

```diff
@@ -360,58 +389,80 @@ namespace AZ::SettingsRegistryMergeUtils
             // InternalScanUpProjectRootKey
-            if (!projectRoot.empty())
+            if (!scanUpProjectRoot.empty())
             {
-                if (projectRoot.IsRelative())
+                if (scanUpProjectRoot.IsRelative())
                 {
-                    if (auto projectAbsPath = AZ::Utils::ConvertToAbsolutePath(projectRoot.Native());
+                    if (auto projectAbsPath = AZ::Utils::ConvertToAbsolutePath(scanUpProjectRoot.Native());
                         projectAbsPath.has_value())
                     {
-                        projectRoot = AZStd::move(*projectAbsPath);
+                        scanUpProjectRoot = AZStd::move(*projectAbsPath);
                     }
                 }
             }
 
-            // Set the {InternalScanUpProjectRootKey} to make sure this code path isn't called again for this settings registry
-            settingsRegistry.Set(InternalScanUpProjectRootKey, projectRoot.Native());
-            if (!projectRoot.empty())
-            {
-                settingsRegistry.Set(projectRootKey, projectRoot.c_str());
-                // Inject the project root at to the front of the command line settings
-                Internal::InjectSettingToCommandLine(settingsRegistry, projectRootKey, projectRoot.Native());
+            // Set the {SetregFileProjectRootKey} to make sure this code path isn't called again for this instance
+            settingsRegistry.Set(Internal::ScanUpProjectRootKey, scanUpProjectRoot.Native());
+        }
+
+        // Check for the project path to used in priority of
+        // 1. command-line
+        // 2. First project.json found by scanning upwards from the current executable directory
+        // 3. "/Amazon/AzCore/Bootstrap/project_path" key set in .setreg file
 
-                return projectRoot;
+        AZ::IO::FixedMaxPath projectRoot;
+
+        // 1. Parse Command Line
+        auto VisitCommandLineOptions = [&projectRoot](const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
+        {
+            constexpr AZStd::string_view ProjectPathOptionName = "project-path";
+            // Lookup the "/O3DE/Runtime/CommandLine/%u/Option" for each command line parameter to see if
+            // the project -path key is available
+            // Check if the option value --project-path has been found key has been found
+            auto cmdProjectPathKey = FixedValueString::format("%.*s/Option", AZ_STRING_ARG(visitArgs.m_jsonKeyPath));
+            if (FixedValueString optionName;
+                visitArgs.m_registry.Get(optionName, cmdProjectPathKey) && optionName == ProjectPathOptionName)
+            {
+                // Updated the existing cmdProjectPathKey to read the value from the command line
+                cmdProjectPathKey = FixedValueString::format("%.*s/Value", AZ_STRING_ARG(visitArgs.m_jsonKeyPath));
+                visitArgs.m_registry.Get(projectRoot.Native(), cmdProjectPathKey);
```

Comment:
Do you want to stop visiting after finding a match?
```suggestion
                visitArgs.m_registry.Get(projectRoot.Native(), cmdProjectPathKey);
                return AZ::SettingsRegistryInterface::VisitResponse::Done;
```

Reply:
Actually no, we want to continue to make sure that the final `--project-path` is used in the case when the `--project-path` is supplied multiple times.
Such as from automated scripts that launch the project-path with default parameters.
i.e
```PS
AssetProcessorBatch.exe --project-path AutomatedTesting ... --project-path D:/Path/To/Other/Project
```

Reply:
A comment has been added indicating that the command line option visitation will continue.